### ISSUE TITLE
[ally]: Combobox Default & with Layer - screen reader disabled option

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#16046]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
Though it is common to remove the disabled element from focus, in this scenario it may be helpful to the user if the disabled option is made available programaticlly with aria-disabled status. Otherwise the SR user might wunder why the option is skipped.

> What happened instead?
Screen reader announcement of disabled option is inconsistent with the non-disabled options

In jaws with FF After expanding the dropdown, when the user navigates through the dropdown options using up/down arrow, After option 2, SR focus moves to option 4 skipping the disabled option 3. 

Note: In VO iPhone the disabled option is announced as "option 3 A disabled item". This inconsistent behaviour may confuse the user.

> What WCAG 2.1 checkpoint does the issue violate?
4.1.2 Name, Role, Value 

## Steps to reproduce the issue
Default: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
With layer: https://react.carbondesignsystem.com/?path=/story/components-combobox--with-layer

> Please create a reduced test case in CodeSandbox
> https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox

## Additional information

- Screenshots or code
- Notes
